pkg/api_builder/lib/api_builder.dart

```diff
@@ -97,12 +97,13 @@ abstract class $utilities {
     T Function(Map<String, dynamic>) fromJson,
   ) async {
     try {
-      // TODO: Consider enforcing that requests should have 'Content-Type' set to
-      // 'application/json'. Notice that we should be careful as the CLI client,
-      // might be sending a different Content-Type.
-      if (!_expectedJsonMimeTypes.contains(request.mimeType)) {
-        _log.info('Unexpected MIME type: ${request.mimeType}', request.mimeType,
-            StackTrace.current);
+      final mimeType = request.mimeType;
```

Comment:
How is `request.mimeType` handled. Specifically, will it parse something like `application/json; charset="utf-8"`.

Reply:
If the type is malformed, accessing the mimeType will throw FormatException, which we catch and reject the request with 400. Otherwise we'll get the media type part of the content type header.